operations/math/addPairWise/operation.go

```diff
@@ -0,0 +1,111 @@
+package addPairWise
+
+import (
+	"fmt"
+
+	"github.com/project-flogo/catalystml-flogo/action/operation"
+
+	"github.com/project-flogo/core/data/coerce"
+
+	"github.com/project-flogo/core/support/log"
+)
+
+type Operation struct {
+	logger log.Logger
+}
+
+func New(ctx operation.InitContext) (operation.Operation, error) {
+	return &Operation{logger: ctx.Logger()}, nil
+}
+
+func (a *Operation) Eval(inputs map[string]interface{}) (interface{}, error) {
+	//To get the inputs in the desired types.
+	input := &Input{}
+
+	input.FromMap(inputs)
+
+	a.logger.Info("Starting operation multPairWise.")
+
+	if inputs["matrix0"] == nil || inputs["matrix1"] == nil {
+
+		var mtx []interface{}
+		if inputs["matrix0"] != nil {
+			mtx = inputs["matrix0"].([]interface{})
```

Comment:
I think using `coerce.ToArray()` should be better. As it can take care of errors if present, what do you think?

Reply:
sounds good, but since you are going to merge all the PairWise together in the implementation and then just change the operation it would be better to change the cast to coerce.ToArray() in the new merged PairWise function.